### PR TITLE
Dịch "mlp-scratch_vn" - Phần 1

### LINKED ISSUE #836: Dịch "mlp-scratch_vn" - Phần 1
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/mlp-scratch_vn.md#L1-L72

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tamnguyenvan**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #836

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-# *dịch tiêu đề phía trên*
+# Cách lập trình Perceptron Đa tầng từ đầu
 :label:`sec_mlp_scratch`
 
 <!--
 Now that we have characterized multilayer perceptrons (MLPs) mathematically, let's try to implement one ourselves.
 -->
 
-*dịch đoạn phía trên*
+Chúng ta đã mô tả perceptron đa tầng (MLPs) ở dạng toán học, giờ hãy cùng lập trình một mạng như vậy xem sao.
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Chúng ta đã mô tả perceptron đa tầng (MLPs) ở dạng toán học, giờ hãy cùng thử lập trình một mạng như vậy xem sao.
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 2]
```diff
@@ -46,15 +46,20 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
 Typically, we choose layer widths in powers of $2$ which tends to be computationally efficient because of how memory is alotted and addressed in hardware.
 -->
 
-*dịch đoạn phía trên*
+Nhớ lại rằng Fashion-MNIST gồm có $10$ nhãn, mỗi ảnh gồm có một lưới $28 \times 28 = 784$ điểm ảnh (đen và trắng).
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Nhắc lại rằng Fashion-MNIST gồm có $10$ nhãn, mỗi ảnh gồm có một lưới $28 \times 28 = 784$ điểm ảnh (đen và trắng).
```
Mình thấy nếu dịch `nhớ lại` thì hơi kỳ. Dù sao tác giả cũng chủ động nhắc cho người đọc

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 3]
```diff
@@ -46,15 +46,20 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
...
+Chúng ta sẽ tiếp tục bỏ qua liên hệ về mặt không gian giữa các điểm ảnh (hiện tại), khi đó ta có thể coi nó đơn giản như một tập dữ liệu phân loại với $784$ đặc trưng đầu vào và $10$ nhãn.
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Chúng ta sẽ tiếp tục bỏ qua liên hệ về mặt không gian giữa các điểm ảnh (ở thời điểm hiện tại), khi đó ta có thể coi nó đơn giản như một tập dữ liệu phân loại với $784$ đặc trưng đầu vào và $10$ lớp.
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 4]
```diff
@@ -46,15 +46,20 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
...
+Để bắt đầu, chúng ta sẽ lập trình một mạng MLP chỉ có một tầng ẩn với $256$ nút ẩn.
+Lưu ý rằng ta có thể coi cả hai đại lượng này là *siêu tham số* và nói chung ta nên thiết lập giá trị cho chúng dựa trên chất lượng trên tập kiểm định.
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Lưu ý rằng ta có thể coi cả hai đại lượng này là các *siêu tham số* và nên thiết lập giá trị cho chúng dựa trên chất lượng trên tập kiểm định. 
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 5]
```diff
@@ -46,15 +46,20 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
 Typically, we choose layer widths in powers of $2$ which tends to be computationally efficient because of how memory is alotted and addressed in hardware.
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]: Phần này chưa dịch nhé bạn

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 6]
```diff
@@ -46,15 +46,20 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
...
+Lưu ý rằng ta có thể coi cả hai đại lượng này là *siêu tham số* và nói chung ta nên thiết lập giá trị cho chúng dựa trên chất lượng trên tập kiểm định. 
 
 <!--
 Again, we will represent our parameters with several `ndarray`s.
 Note that *for every layer*, we must keep track of one weight matrix and one bias vector.
 As always, we call `attach_grad` to allocate memory for the gradients (of the loss) with respect to these parameters.
 -->
 
-*dịch đoạn phía trên*
+Chúng ta sẽ lại biễu diện các tham số bằng một số `ndarray`.
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Chúng ta sẽ lại biễu diện các tham số bằng một vài `ndarray`.
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 7]
```diff
@@ -46,15 +46,20 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
...
+Lưu ý rằng *với mỗi tầng*, ta luôn có một ma trận trọng số và một vector hiệu chỉnh.
```
Comment (rootonchair) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]: hmm, từ `must keep track` nên dịch như thế nào nhỉ? Dịch là `phải tính đạo hàm` có được không?

**Comment by duythanhvn** [comment]
Update em ơi.

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md (hunk 7 above)
Comment (rootonchair) [reply chapter_multilayer-perceptrons/mlp-scratch_vn.md]: À, cũng hợp lý. Mình lúc đầu lại nghĩ `keep track` ở đây kiểu giống như đánh dấu cho thư viện để ý tới weight và bias nhằm lấy đạo hàm sau này

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 8]
```diff
@@ -5,14 +5,14 @@
 # Implementation of Multilayer Perceptron from Scratch
 -->
 
-# *dịch tiêu đề phía trên*
+# Cách lập trình Perceptron Đa tầng từ đầu
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
# Lập trình Perceptron Đa tầng từ đầu
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 9]
```diff
@@ -36,7 +36,7 @@ train_iter, test_iter = d2l.load_data_fashion_mnist(batch_size)
 ## Initializing Model Parameters
 -->
 
-## *dịch tiêu đề phía trên*
+## Khởi tạo tham số mô hình
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
## Khởi tạo Tham số Mô hình
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 10]
```diff
@@ -46,15 +46,21 @@ Note that we can regard both of these quantities as *hyperparameters* and ought
 Typically, we choose layer widths in powers of $2$ which tends to be computationally efficient because of how memory is alotted and addressed in hardware.
 -->
 
-*dịch đoạn phía trên*
+Nhắc lại rằng Fashion-MNIST gồm có $10$ nhãn, mỗi ảnh gồm có một lưới $28 \times 28 = 784$ điểm ảnh (đen và trắng).
+Chúng ta sẽ tiếp tục bỏ qua liên hệ về mặt không gian giữa các điểm ảnh (ở thời điểm hiện tại), khi đó ta có thể coi nó đơn giản như một tập dữ liệu phân loại với $784$ đặc trưng đầu vào và $10$ lớp.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Chúng ta sẽ tiếp tục bỏ qua liên hệ về mặt không gian giữa các điểm ảnh (ở thời điểm hiện tại), khi đó ta có thể xem nó đơn giản như một tập dữ liệu phân loại với $784$ đặc trưng đầu vào và $10$ lớp.
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md [hunk 11]
```diff
@@ -25,7 +25,7 @@ To compare against our previous results achieved with (linear) softmax regressio
 we will continue work with the Fashion-MNIST image classification dataset (:numref:`sec_fashion_mnist`).
 -->
 
-*dịch đoạn phía trên*
+Để so sánh với kết quả đã đạt được trước đó bằng hồi quy softmax (tuyến tính) (:numref:`sec_softmax_scratch`), chúng ta sẽ tiếp tục sử dụng tập dữ liệu phân loại ảnh Fashion-MNIST.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/mlp-scratch_vn.md]:
```suggestion
Để so sánh với kết quả đã đạt được trước đó bằng hồi quy softmax -- tuyến tính (:numref:`sec_softmax_scratch`), chúng ta sẽ tiếp tục sử dụng tập dữ liệu phân loại ảnh Fashion-MNIST.
```

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md (hunk 10 above)
Comment (aivivn) [reply chapter_multilayer-perceptrons/mlp-scratch_vn.md]: `coi` là `consider`, `xem` là `see` :)

**Comment by tamnguyenvan** [comment]
@duythanhvn @rootonchair done.

File: chapter_multilayer-perceptrons/mlp-scratch_vn.md (hunk 10 above)
Comment (duythanhvn) [reply chapter_multilayer-perceptrons/mlp-scratch_vn.md]: @tiepvupsu @aivivn Anh có thể nói kỹ hơn ý của anh trong câu chuyện này không ạ? Em nghĩ em chưa rõ được ý của anh ạ.

Hai từ này trong tiếng Việt là từ đồng nghĩa với ngữ nghĩa trong phần lớn ngữ cảnh là tương đồng nhau. Và hai từ này còn là từ đơn nữa, nên `consider` nếu dịch là `coi` không thì không hề đủ nghĩa; khi đặt trong ngữ cảnh kỹ thuật nếu tìm một từ mang nghĩa `consider` thì có thể tìm tới từ cân nhắc, suy nghĩ kỹ, chú ý đến hoặc là xem xét... nó là từ ghép.

Với từ `coi` em thấy từ mặc dù đồng nghĩa nhưng giới hạn sử dụng của nó lại không rộng bằng `xem`, và cũng vì nó là từ đơn, đồng nghĩa nên em mới có ý chọn từ để thống nhất hơn khi lựa chọn sử dụng.

Comment (tiepvupsu) [reply chapter_multilayer-perceptrons/mlp-scratch_vn.md]: Thực ra ngày đi học anh hay dùng từ `coi` khi làm toán. Ít khi dùng từ `xem` mặc dù anh là người Bắc.

Từ `consider` vừa có nghĩa là `xem xét` vừa có thể hiểu là `coi như`. Trong trường hợp này nó mang nghĩa thứ hai. 

Anh vote giữ nguyên từ `coi`. Dù sao thì từ này ở đây cũng không quan trọng lắm.
